src/runtime/join.ts

```diff
@@ -218,6 +218,109 @@ export class Scan {
     return foo.join("");
   }
 
+  
+  // // I can write macros with eval, hmm
```

Comment:
I think that if this commented code is part of other work, it should be included in a separate PR. For now just take the commented code out of this PR.